Tiny PR: update default providers list (#2574)
astronomer/docs: astro/runtime-image-architecture.md

## Provider packages

> The latest version of the Astro Runtime image has the following open source provider packages pre-installed. Providers marked with an asterisk (`*`) are installed only in Astro Runtime and not installed by default on Apache Airflow.

Reviewer: ```suggestion
The latest version of the Astro Runtime image has the following open source provider packages pre-installed. Providers marked with an asterisk (*) are installed only in Astro Runtime and not installed by default on Apache Airflow.
```

> - Amazon [`apache-airflow-providers-amazon`](https://pypi.org/project/apache-airflow-providers-amazon/) `*`

Reviewer: ```suggestion
- Amazon [`apache-airflow-providers-amazon`](https://pypi.org/project/apache-airflow-providers-amazon/)*
```
Oh maybe the code styling was confusing on my part, I think we should just use plain asterisks in the list itself.

> - Astronomer Providers [`astronomer-providers`](https://pypi.org/project/astronomer-providers/) *

Reviewer: ```suggestion
- Astronomer Providers [`astronomer-providers`](https://pypi.org/project/astronomer-providers/)*
```
Can we remove the spaces on these pls 🙈

- Celery [`apache-airflow-providers-celery`](https://pypi.org/project/apache-airflow-providers-celery/)
- Cloud Native Computing Foundation (CNCF) Kubernetes [`apache-airflow-providers-cncf-kubernetes`](https://pypi.org/project/apache-airflow-providers-cncf-kubernetes/)
- Common SQL [`apache-airflow-providers-common-sql`](https://pypi.org/project/apache-airflow-providers-common-sql/)
- Datadog [`apache-airflow-providers-datadog](https://pypi.org/project/apache-airflow-providers-datadog/) `*`
- Elasticsearch [`apache-airflow-providers-elasticsearch`](https://pypi.org/project/apache-airflow-providers-elasticsearch/) `*`
- FTP [`apache-airflow-providers-ftp`](https://pypi.org/project/apache-airflow-providers-ftp/) 
- Google [`apache-airflow-providers-google`](https://pypi.org/project/apache-airflow-providers-google/) `*`
- HTTP [`apache-airflow-providers-http`](https://pypi.org/project/apache-airflow-providers-http/)
- IMAP [`apache-airflow-providers-imap`](https://pypi.org/project/apache-airflow-providers-imap/)
- Microsoft Azure [`apache-airflow-providers-microsoft-azure`](https://pypi.org/project/apache-airflow-providers-microsoft-azure/) `*`

> - OpenLineage [`openlineage-airflow`](https://pypi.org/project/openlineage-airflow/) (this package installs an Airflow plugin) `*`

Reviewer: ```suggestion
- OpenLineage [`openlineage-airflow`](https://pypi.org/project/openlineage-airflow/) (Installed through an Airflow plugin) `*`
```
Would this be accurate?

Author: unfortunately no. The package itself is for a plugin, that is what is being installed. 😅

## Provider packages

> The latest version of the Astro Runtime image has the following open source provider packages pre-installed. Providers marked with (pre-installed OSS) are pre-installed in open-source Airflow.

Reviewer: ```suggestion
The latest version of the Astro Runtime image has the following open source provider packages pre-installed. Providers marked with an asterisk (`*`) are installed only in Astro Runtime and not installed by default on Apache Airflow.
```
Wouldn't a user care more about what packages are preinstalled on Runtime, but not OSS? Because of that, I would flip this around.

Author: That is a very good point!

- Celery [`apache-airflow-providers-celery`](https://pypi.org/project/apache-airflow-providers-celery/) (pre-installed OSS)
- Cloud Native Computing Foundation (CNCF) Kubernetes [`apache-airflow-providers-cncf-kubernetes`](https://pypi.org/project/apache-airflow-providers-cncf-kubernetes/) (pre-installed OSS)
- Common SQL [`apache-airflow-providers-common-sql`](https://pypi.org/project/apache-airflow-providers-common-sql/) (pre-installed OSS)
- Datadog [`apache-airflow-providers-datadog](https://pypi.org/project/apache-airflow-providers-datadog/)
- Elasticsearch [`apache-airflow-providers-elasticsearch`](https://pypi.org/project/apache-airflow-providers-elasticsearch/)
- FTP [`apache-airflow-providers-ftp`](https://pypi.org/project/apache-airflow-providers-ftp/) (pre-installed OSS)
- Google [`apache-airflow-providers-google`](https://pypi.org/project/apache-airflow-providers-google/)
- HTTP [`apache-airflow-providers-http`](https://pypi.org/project/apache-airflow-providers-http/) (pre-installed OSS)
- IMAP [`apache-airflow-providers-imap`](https://pypi.org/project/apache-airflow-providers-imap/) (pre-installed OSS)

> - Microsoft Azure [`apache-airflow-providers-microsoft-azure`](https://pypi.org/project/apache-airflow-providers-microsoft-azure/)

Reviewer: ```suggestion
- Microsoft Azure [`apache-airflow-providers-microsoft-azure`](https://pypi.org/project/apache-airflow-providers-microsoft-azure/)
- OpenLineage with Airflow [`openlineage-airflow`](https://pypi.org/project/openlineage-airflow/)
```

- PostgreSQL (Postgres) [`apache-airflow-providers-postgres`](https://pypi.org/project/apache-airflow-providers-postgres/)
- Redis [`apache-airflow-providers-redis`](https://pypi.org/project/apache-airflow-providers-redis/)
- SQLite [`apache-airflow-providers-sqlite`](https://pypi.org/project/apache-airflow-providers-sqlite/) (pre-installed OSS)

> Additionally, the Astro Runtime image contains a plugin to interact with [OpenLineage](https://pypi.org/project/openlineage-airflow/). 

Reviewer: Is the end result the same? As in, when you scan the images for dependencies, will OL appear? If so, I'm not sure it's worth distinguishing that this is installed at the plugin level. And even if so, I would add it to the bullet list with a parentheses addendum: `(installed through a separate plugin)` or something similar
